app/scripts/models/auth_brokers/fx-desktop.js

```diff
@@ -65,6 +71,36 @@ define([
           self, options);
     },
 
+    /**
+      * Is used to ask the client about capabilities before rendering the views
+      *
+      * @returns {Promise}
+      */
+    beforeRender: function () {
+      var self = this;
+      var defer = p.defer();
+      var result = null;
+
+      self.request(self._commands.CAPABILITIES)
```

Comment:
:+1: I like the single call idea as well. Is there an advantage to calling on every page view?